plugins/module_utils/api.py

```diff
@@ -262,6 +264,25 @@ def create(self, resource, data=None):
             resource['state'] = "present"
         return resource
 
+    def wait_for_state(self, check_parameter, allowed_states):
+        start = datetime.now()
+        timeout = self._module.params['api_timeout'] * 2
+        while datetime.now() - start < timedelta(seconds=timeout):
+            info = self.query()
+            if info.get(check_parameter) in allowed_states:
+                return info
+            sleep(1)
+
+        # Timeout succeeded
+        if info.get('name') is not None:
+            msg = "Timeout while waiting for a state change on load balancer %s to states %s. " \
+                    "Current state is %s." % (info.get('name'), allowed_states, info.get('state'))
+        else:
+            name_uuid = self._module.params.get('name') or self._module.params.get('uuid')
+            msg = 'Timeout while waiting to find the load balancer %s' % name_uuid
```

Comment:
This text seems wrong. And instead of having an if / else clause you could just do:
`name_uuid = info.get('name') or self._module.params.get('name') or self._module.params.get('uuid')`

But after thinking about it a bit more, I wonder why `wait_for_state` can be called in cases where the resource might not (yet) exist.